syntax = "proto2";

package senior.dox.proto;

message MonopolyTask {
  optional string name = 1;
  optional uint32 prio = 2 [default = 1];
  optional string group_name = 3;
}

message Sched2Group {
  required string name = 1 [default = "default_grp"];
  optional uint32 processor_num = 2;
  optional string affinity = 3;
  optional string cpuset = 4;
  optional string processor_policy = 5;
  optional int32 processor_prio = 6 [default = 0];
  repeated MonopolyTask tasks = 7;
}

message MonopolyConf {
  repeated Sched2Group groups = 1;
}
